### ISSUE TITLE
Invalid JSON RPC 2.0 error responses

### ISSUE THREAD

**Issue by mfya** [cenobites/flask-jsonrpc#29]
Some errors cause response objects that violate the JSON RPC 2.0 spec, because they have both, a "result" and "error" member set. An example would be MethodNotFoundError.

See https://github.com/cenobites/flask-jsonrpc/blob/723db17b1ef11d15d4180fd0e372b9a51723f4e4/flask_jsonrpc/site.py#L186-195 .
It seems that the version is initialized to late in the marked block and early exceptions are handled with the JSON RPC 1.0 spec, because the version_hint is always "1.0".

**Comment by nycholas** [comment]
Hello,

By default the version is "1.0", in line[1] remove key "error" by response, and in lines[2][3][4] 
remove key "result" by response if throw error.


[1] - https://github.com/cenobites/flask-jsonrpc/blob/723db17b1ef11d15d4180fd0e372b9a51723f4e4/flask_jsonrpc/site.py#L253
[2] - https://github.com/cenobites/flask-jsonrpc/blob/723db17b1ef11d15d4180fd0e372b9a51723f4e4/flask_jsonrpc/site.py#L234
[3] - https://github.com/cenobites/flask-jsonrpc/blob/723db17b1ef11d15d4180fd0e372b9a51723f4e4/flask_jsonrpc/site.py#L241
[4] - https://github.com/cenobites/flask-jsonrpc/blob/723db17b1ef11d15d4180fd0e372b9a51723f4e4/flask_jsonrpc/site.py#L248

**Comment by nycholas** [comment]
Hello,

```
$ curl -i -X POST -d '{"jsonrpc": "2.0", "method": "App.methodNotFound", "params": {}, "id": "1"}' http://localhost:5000/api
HTTP/1.0 200 OK
Content-Type: application/json
Content-Length: 515
Server: Werkzeug/0.8.3 Python/2.7.9
Date: Thu, 16 Apr 2015 13:45:44 GMT

{
  "error": {
    "code": -32601, 
    "data": null, 
    "executable": "/usr/bin/python2", 
    "message": "MethodNotFoundError: Method not found. Available methods: App.index\nsystem.describe", 
    "name": "MethodNotFoundError", 
    "stack": "Traceback (most recent call last):\n  File \"/home/nycholas/project/src/o_lalertom/flask/flask-jsonrpc/flask_jsonrpc/site.py\", line 184, in response_dict\n    .format('\\n'.join(list(self.urls.keys()))))\nMethodNotFoundError\n"
  }, 
  "id": "1", 
  "result": null
```

Yes! It's a problem.

Thank you.

**Comment by nycholas** [comment]
Fixed in version 0.3.

Thank you.
